LedDashboard/Modules/LeagueOfLegends/ChampionModules/AzirModule.cs

```diff
@@ -0,0 +1,136 @@
+using LedDashboard.Modules.BasicAnimation;
+using LedDashboard.Modules.Common;
+using LedDashboard.Modules.LeagueOfLegends.ChampionModules.Common;
+using LedDashboard.Modules.LeagueOfLegends.Model;
+using SharpDX.RawInput;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace LedDashboard.Modules.LeagueOfLegends.ChampionModules
+{
+    [Champion(CHAMPION_NAME)]
+    class AzirModule : ChampionModule
+    {
+
+        // Change to whatever champion you want to implement
+        public const string CHAMPION_NAME = "Azir";
+
+        // Variables
+
+        // Champion-specific Variables
+
+
+        /// <summary>
+        /// Creates a new champion instance.
+        /// </summary>
+        /// <param name="ledCount">Number of LEDs in the strip</param>
+        /// <param name="gameState">Game state data</param>
+        /// <param name="preferredLightMode">Preferred light mode</param>
+        /// <param name="preferredCastMode">Preferred ability cast mode (Normal, Quick Cast, Quick Cast with Indicator)</param>
+        public static AzirModule Create(int ledCount, GameState gameState, LightingMode preferredLightMode, AbilityCastPreference preferredCastMode = AbilityCastPreference.Normal)
+        {
+            return new AzirModule(ledCount, gameState, CHAMPION_NAME, preferredLightMode, preferredCastMode);
+        }
+
+
+        private AzirModule(int ledCount, GameState gameState, string championName, LightingMode preferredLightMode, AbilityCastPreference preferredCastMode)
+                            : base(ledCount, championName, gameState, preferredLightMode)
+        {
+            // Initialization for the champion module occurs here.
+
+            // Set preferred cast mode. It's a player choice (Quick cast, Quick cast with indicator, or Normal cast)
+            PreferredCastMode = preferredCastMode;
+
+            // Set cast modes for abilities.
+            // For Vel'Koz, for example:
+            // Q -> Normal ability, but it can be recast within 1.15s
+            // W -> Normal ability
+            // E -> Normal ability
+            // R -> Instant ability, it is cast the moment the key is pressed, but it can be recast within 2.3s
+            Dictionary<AbilityKey, AbilityCastMode> abilityCastModes = new Dictionary<AbilityKey, AbilityCastMode>()
+            {
+                [AbilityKey.Q] = AbilityCastMode.Instant(),
+                [AbilityKey.W] = AbilityCastMode.Instant(),
+                [AbilityKey.E] = AbilityCastMode.Instant(),
+                [AbilityKey.R] = AbilityCastMode.Instant(),

```

Comment:
Q,W, and R have Normal cast modes, because they are cast by pressing the key and then clicking.